openwisp_controller/config/tasks_zerotier.py

```
@@ -91,43 +50,40 @@ def handle_api_call(self, fn, *args, send_notification=True, **kwargs):
+            if status_code is None or status_code in self._RECOVERABLE_API_CODES:
+                retry_logger = logger.warning
                 # When retry limit is reached, use error logging
                 if self.request.retries == self.max_retries:
                     retry_logger = logger.error
                 retry_logger(
                     f"Try [{self.request.retries}/{self.max_retries}] "
                     f"{err_msg}, Error: {e}"
                 )
-                raise e
+                raise
             logger.error(f"{err_msg}, Error: {e}")
```

Ruff (0.14.14)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)